### DISCUSSION TITLE
How can i turn no data to zero in Loki

### DISCUSSION THREAD

**Discussion by waterdrop01** [grafana/grafana#39742] [Alerting]
I open this discussion based on this thread => https://community.grafana.com/t/how-can-i-turn-no-data-to-zero-in-loki/40694

My Loki alert is counting the occurence of the work "error" in the logs. Most of the time, this query (logql) return `NO DATA`.

As I'm using a template function in the annotation fields, Grafana throws the following error:

```
t=2021-09-28T15:20:03+0000 lvl=eror msg="error in expanding template" logger=ngalert name=errorvalue value="{{ $values.B }}" err="error executing template __alert_PROD log errors: template: __alert_PROD log errors:1:86: executing \"__alert_PROD log errors\" at <$values.B>: map has no entry for key \"B\""
t=2021-09-28T15:20:03+0000 lvl=eror msg="error in expanding template" logger=ngalert name=object value="Service: {{ $values.B.Labels.app }}" err="error executing template __alert_PROD log errors: template: __alert_PROD log errors:1:95: executing \"__alert_PROD log errors\" at <$values.B.Labels.app>: map has no entry for key \"B\""
```

Having a way to set the timeserie to 0 would probably solve this error?

**Comment by swiffer** [comment]
The Link is suggesting to do a transformation that did not work out in my case (multiple series / streams).

This is what I did:

![grafik](https://user-images.githubusercontent.com/2990373/143475491-a64e86b4-9dcd-4e0a-a6b8-6669c9627e39.png)

additionally this seems to be adressed here: https://github.com/grafana/loki/issues/4420

**Comment by liguozhong** [comment]
https://github.com/grafana/loki/pull/7023
fix this
